datadog_lambda/tracing.py

```diff
@@ -313,19 +323,35 @@ def _extract_context_from_eventbridge_sqs_event(event):
     body = json.loads(body_str)
     detail = body.get("detail")
     dd_context = detail.get("_datadog")
+
+    if is_step_function_event(dd_context):
+        try:
+            return extract_context_from_step_functions(dd_context, None)
+        except Exception:
+            logger.debug(
+                "Failed to extract Step Functions context from EventBridge to SQS event."
+            )
+
     return propagator.extract(dd_context)
 
 
 def extract_context_from_eventbridge_event(event, lambda_context):
     """
     Extract datadog trace context from an EventBridge message's Details.
     This is only possible if Details is a JSON string.
+
+    If we find a Step Function context, try to extract the trace context from
+    that header.
     """
     try:
         detail = event.get("detail")
         dd_context = detail.get("_datadog")
         if not dd_context:
             return extract_context_from_lambda_context(lambda_context)
+
+        if is_step_function_event(dd_context):
+            return extract_context_from_step_functions(dd_context, lambda_context)
```

Comment:
This one isn't wrapped in a try/except, but the two above are, why is that?

Reply:
Ah I also meant to wrap this in try/except but lemme explain why I'm doing it

I wanted to pass in `lambda_context=None` so that if the extractor fails, it won't fallback on the `lambda_context` and it'll continue with the normal codepath and call
```
return propagator.extract(dd_context)
```